Code Insights: Add explanation tooltip on the creation insight form.

vovakulikov opened the pull request:
Closes https://github.com/sourcegraph/sourcegraph/issues/22963

### Background
This PR adds an explanation tooltip for the disabled global visibility radio button in the creation form and fixes tooltip dashboard-related texts. 

Also, this PR adds a changelog announcement line about all repos insight

vovakulikov commented on CHANGELOG.md:
```diff
@@ -27,8 +27,8 @@ All notable changes to Sourcegraph are documented in this file.
 - Code Insights can now be grouped into dashboards. [#22215](https://github.com/sourcegraph/sourcegraph/issues/22215)
-- Added the code insights dashboards functionality. Now you can divide insights into dashboards. [#22215](https://github.com/sourcegraph/sourcegraph/issues/22215)
 - Batch Changes changesets can now be [published from the Sourcegraph UI](https://docs.sourcegraph.com/batch_changes/how-tos/publishing_changesets#within-the-ui). [#18277](https://github.com/sourcegraph/sourcegraph/issues/18277)
+- Experimental: Search-based code insights can run over all repositories on the instance. To enable, use the feature flag `"experimentalFeatures": { "codeInsightsAllRepos": true }`. [#22759](https://github.com/sourcegraph/sourcegraph/issues/22759)
```
@felixfbecker Can you please take a look at this line

felixfbecker replied:
```suggestion
- Experimental: Search-based code insights can run over all repositories on the instance. To enable, use the feature flag `"experimentalFeatures": { "codeInsightsAllRepos": true }` and tick the checkbox in the insight creation/edit UI. [#22759](https://github.com/sourcegraph/sourcegraph/issues/22759)
```

vovakulikov commented on client/web/src/insights/pages/dashboards/creation/components/insights-dashboard-creation-content/utils/get-global-subject-tooltip-text.ts:
```diff
@@ -10,7 +10,7 @@ export function getGlobalSubjectTooltipText(globalSubject: SettingsSiteSubject |
     const globalSubjectAdminCheckMessage = globalSubject.viewerCanAdminister
         ? undefined
-        : "You don't have a permission to change global scope. Reach out your site admin"
+        : "You don't have permission to change the global scope. Reach out to your site admin"
```
@Joelkw @AlicjaSuska @felixfbecker these tooltip texts require proofreading.

felixfbecker replied:
I might just shorten/combine the sentences:
```suggestion
        : "Only site admins can change global settings"
```
Ideally I'd say "Only site admins can define global insights" or "[...] dashboards" if we know here whether this is for an insight or dashboard.

AlicjaSuska reviewed (changes requested):
@vovakulikov as we've discussed today, we should change texts a little so they are more understandable for the users:

Creation UI:
- 'Only site admins can create global insights'.
I think it is more in the context of creation UI than saying 'Only site admins can change global settings' - as user is not directly trying to change the settings


Default dashboards:
- For orgs: 'This is an automatically created dashboard that lists all the [Org]'s insights. You can't edit this dashboard.' - this would explicitly state why editing is not available
- For All insights: 'This is an automatically created dashboard that lists all the  insights you have access to. You can't edit this dashboard.'
- For all private: 'This is an automatically created dashboard that lists all your private insights. You can't edit this dashboard.'

vovakulikov commented:
@AlicjaSuska I changed all text according to your comment above, thanks for it btw. 

Also, I used assign instead of `add` in the dashboard initial state and changed Add/remove insights to Assign or Remove insights in the dashboard context menu action.

AlicjaSuska commented on client/web/src/insights/pages/dashboards/dashboard-page/hooks/use-dashboard-permissions.ts:
```diff
@@ -85,8 +93,18 @@ export function getTooltipMessage(permissions: DashboardPermissions): string | u
+                    return "This is an automatically created dashboard that lists all the insights you have access to. You can't edit this dashboard."
+                case InsightsDashboardType.Personal:
+                    return "This is an automatically created dashboard that lists all your private insights. You can't edit this dashboard."
+                case InsightsDashboardType.Organization:
```
@vovakulikov can we add :' This is an automatically created dashboard that lists all the [${dashboard.organisation.name} insights. You can't edit this dashboard.' for the Orgs?

vovakulikov replied:
@AlicjaSuska not sure I got it. What is the difference between the current text and the text you suggested? dashboard owner name has a value of organization name in case if that dashboard has org scope of visibility if you meant it.

AlicjaSuska replied:
Oh, got it now. Sorry, please ignore the comment above. Thank you!

AlicjaSuska commented on client/web/src/insights/pages/dashboards/dashboard-page/hooks/use-dashboard-permissions.ts:
```diff
@@ -85,8 +93,18 @@ export function getTooltipMessage(permissions: DashboardPermissions): string | u
+                    return "This is an automatically created dashboard that lists all your private insights. You can't edit this dashboard."
+                case InsightsDashboardType.Organization:
+                case InsightsDashboardType.Global:
+                    return `This is an automatically created dashboard that lists all the ${dashboard.owner.name}'s insights. You can't edit this dashboard.`
```
@vovakulikov in the current implementation we use '[user] insights' not '[users]'s insight'. I think we should be consistent here as well and change to ${dashboard.owner.name} insights (no 's). It came from my previous comment, sorry for this mistake. Thank you!

vovakulikov replied:
@AlicjaSuska done, thanks

felixfbecker replied:
I'd actually prefer if we changed the implementation to use possessive form everywhere, that's what we decided on in the design. The possessive form communicates the concept of ownership better and "Vova insights" sounds wrong grammatically imo.
